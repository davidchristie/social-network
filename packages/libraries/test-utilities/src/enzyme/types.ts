import { ReactWrapper, ShallowWrapper } from "enzyme";

export type Wrapper = ReactWrapper | ShallowWrapper;
