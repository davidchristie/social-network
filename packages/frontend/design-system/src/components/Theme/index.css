@import url('https://fonts.googleapis.com/css?family=Lato');

.Theme {
  --theme-header-height: 80px;
  --theme-primary: rgb(40, 60, 75);
  --theme-primary-contrast: white;
  --theme-secondary: rgb(230, 230, 230);
  --theme-secondary-contrast: black;
  font-family: 'Lato', sans-serif;
}

.Theme a {
  color: inherit;
}
