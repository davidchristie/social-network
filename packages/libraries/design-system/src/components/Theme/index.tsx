import createMuiTheme from "@material-ui/core/styles/createMuiTheme";
import MuiThemeProvider from "@material-ui/core/styles/MuiThemeProvider";
import React from "react";

import "./index.css";

const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
  },
});

export default class Theme extends React.Component {
  public render () {
    return (
      <MuiThemeProvider theme={theme}>
        <div className="Theme">
          {this.props.children}
        </div>
      </MuiThemeProvider>
    );
  }
}
