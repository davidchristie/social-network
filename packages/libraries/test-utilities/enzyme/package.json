{
  "main": "../dist/enzyme/index.js",
  "name": "test-utilities/enzyme",
  "private": true,
  "version": "1.0.0"
}
