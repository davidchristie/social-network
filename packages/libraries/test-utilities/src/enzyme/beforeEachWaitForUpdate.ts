import { Wrapper } from "./types";

export default function beforeEachWaitForUpdate (getWrapper: () => Wrapper) {
  beforeEach(async () => {
    await new Promise(resolve => window.setTimeout(resolve, 0));
    getWrapper().update();
  });
}
