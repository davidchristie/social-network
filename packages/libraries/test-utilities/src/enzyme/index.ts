export { default as beforeEachWaitForUpdate } from "./beforeEachWaitForUpdate";
export { default as itContainsComponent } from "./itContainsComponent";
