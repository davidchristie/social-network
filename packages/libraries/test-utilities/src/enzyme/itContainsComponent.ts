import React from "react";
import { Wrapper } from "./types";

export default function itContainsComponent (
  name: string,
  getWrapper: () => Wrapper,
  Component: React.ComponentType
) {
  it(name, () => {
    expect(getWrapper().find(Component).exists()).toBe(true);
  });
}
