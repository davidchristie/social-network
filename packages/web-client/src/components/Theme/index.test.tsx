import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

import Theme from "./index";

describe("Theme component", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Theme />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
