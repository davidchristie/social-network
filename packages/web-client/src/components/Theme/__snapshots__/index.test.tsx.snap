// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme component matches snapshot 1`] = `
<div
  className="Theme"
/>
`;
